druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/query/lookup/dynamic/schema/DynamicLookupFlatbufferSchemaSerDe.java

```diff
@@ -1,48 +1,86 @@
 package com.yahoo.maha.maha_druid_lookups.query.lookup.dynamic.schema;
 
 
-import com.fasterxml.jackson.databind.JsonNode;
 import com.yahoo.maha.maha_druid_lookups.query.lookup.*;
 import com.yahoo.maha.maha_druid_lookups.query.lookup.dynamic.*;
 import com.yahoo.maha.maha_druid_lookups.query.lookup.namespace.*;
+import org.apache.druid.java.util.common.logger.Logger;
 
+import java.nio.ByteBuffer;
 import java.util.*;
+import java.util.stream.Collectors;
+
 
 public class DynamicLookupFlatbufferSchemaSerDe implements  DynamicLookupCoreSchema {
+    private static final Logger LOG = new Logger(DynamicLookupFlatbufferSchemaSerDe.class);
+
+    private DynamicLookupSchema dynamicLookupSchema;
+    private String fieldsCsv;
 
-    //WIP
-    private JsonNode coreSchema;
 
-    public DynamicLookupFlatbufferSchemaSerDe(JsonNode coreSchema){
-        this.coreSchema = coreSchema;
+
+    public DynamicLookupFlatbufferSchemaSerDe(DynamicLookupSchema dynamicLookupSchema){
+        this.dynamicLookupSchema = dynamicLookupSchema;
+        fieldsCsv = dynamicLookupSchema.getSchemaFieldList().stream().map(s-> s.getField()).collect(Collectors.joining(", "));
     }
 
     public ExtractionNameSpaceSchemaType getSchemaType(){
         return ExtractionNameSpaceSchemaType.FLAT_BUFFER;
     }
 
-    public String getSchemaPath() {
-        return "";
-    }
 
     @Override
     public String getValue(String fieldName, byte[] dataBytes, Optional<DecodeConfig> decodeConfigOptional, RocksDBExtractionNamespace extractionNamespace) {
-        // get the index for field
         // init Generic Table from DataBytes; GenericFlatBufferTable
-        // getValue with index;
+        GenericFlatBufferTable genericFlatBufferTable = getDynamicFlatbufferTable(dataBytes);
+
         // handleDecode;
-        return null;
+        if(decodeConfigOptional.isPresent()){
+            return handleDecode(decodeConfigOptional.get(), genericFlatBufferTable, extractionNamespace);
+        } else {
+            return getValueForField(fieldName, genericFlatBufferTable, extractionNamespace);
+        }
+    }
+
+
+    // get the index for field
+    private Optional<Integer> getSchemaFieldIndex(String fieldName){
+        List<SchemaField> fieldList = dynamicLookupSchema.getSchemaFieldList();
+        Optional<SchemaField> schemaField = fieldList.stream().filter(e -> fieldName.equals(e.getField())).findFirst();
```

Comment:
instead of going through the list everytime and filtering on it, lets create the hashmap of FieldName to index from schema field list so that lookup can be done in o(1)